### PR TITLE
Add complete step function

### PR DESCRIPTION
**Pull request by kornel-filep**
## What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)
Introduces the final SF to the CF Template.
Adds the necessary steps to the deployment script.
Adds the SNS Topic to the CF Template.

## What is the current behavior? (You can also link to an open issue here)
A basic step function is created in AWS.

## What is the new behavior (if this is a feature change)?
The planned step function is created in AWS. See the diagram in README.md

## Does this PR introduce a breaking change? (What changes might users need to make in their application due to this PR?)
-

## Other information:
-

### CODE REVIEW

File: deploy.sh [hunk 1]
```diff
@@ -1,19 +1,27 @@
 set -e
 
-aws cloudformation deploy --template-file aws/cf-template.yml --stack-name image-processing-demo --capabilities CAPABILITY_NAMED_IAM
+stateDefinition=$(<aws/state-machine.asl.json)
```
Comment (szikszail) [inline deploy.sh]:
```suggestion
stateDefinition=$(cat aws/state-machine.asl.json | tr -d '\n')
```

File: deploy.sh [hunk 2]
```diff
@@ -1,19 +1,27 @@
...
+
+noNewLineStateDefinition=$(echo "$stateDefinition" | tr -d '\n')
+
+sed -e "s/STATE_FUNCTION_ASL_JSON/$(echo $noNewLineStateDefinition)/" aws/cf-template.yml > aws/cf-template-tmp.yml
```
Comment (szikszail) [inline deploy.sh]:
```suggestion
sed -e "s/STATE_FUNCTION_ASL_JSON/$stateDefinition/" aws/cf-template.yml > aws/cf-template-tmp.yml
```

File: deploy.sh [hunk 3]
```diff
@@ -1,19 +1,27 @@
 set -e
 
-aws cloudformation deploy --template-file aws/cf-template.yml --stack-name image-processing-demo --capabilities CAPABILITY_NAMED_IAM
+stateDefinition=$(<aws/state-machine.asl.json)
+
+noNewLineStateDefinition=$(echo "$stateDefinition" | tr -d '\n')
```
Comment (szikszail) [inline deploy.sh]:
```suggestion

```

**Review by szikszail — CHANGES_REQUESTED** [review]
Just a small suggestion for the shell script

**Comment by kornel-filep** [comment]
thanks @szikszail, added the changes. tbh, forgot about this PR 😅

**Review by szikszail — APPROVED** [review]

**Pull request merged**